src/odb/src/db/dbSite.cpp

```diff
@@ -230,6 +232,62 @@ void dbSite::setClass(dbSiteClass type)
   site->_flags._class = type.getValue();
 }
 
+void dbSite::setRowPattern(
+    std::vector<std::pair<std::string, std::string>>& row_pattern)
+{
+  _dbSite* site = (_dbSite*) this;
+  site->_row_patterns.reserve(row_pattern.size());
+  site->_flags._is_hybrid = 1;
+  std::unordered_map<std::string, dbOrientType::Value> orientationMap
+      = {{"N", dbOrientType::R0},
+         {"W", dbOrientType::R270},
+         {"S", dbOrientType::R180},
+         {"E", dbOrientType::R90},
+         {"FN", dbOrientType::MYR90},
+         {"FW", dbOrientType::MY},
+         {"FS", dbOrientType::MXR90},
+         {"FE", dbOrientType::MX}};
+
+  for (auto& row : row_pattern) {
+    dbOrientType orient = dbOrientType::R0;
+    auto _site = (_dbSite*) getLib()->findSite(row.first.c_str());
+    _site->_flags._is_hybrid = 1;
+    auto it = orientationMap.find(row.second);
+    if (it != orientationMap.end()) {
```

Comment:
If not found it should be an error

Reply:
there is a possibility that it is not constructed yet.
It depends on the order of parsing and finding rows and row patterns, so it might not be an issue here. It will be flagged as an error in other places in the code, but here it is not an issue (yet).